src/marks/arrow.js

```diff
@@ -76,8 +76,9 @@ export class Arrow extends Mark {
               // The start ⟨x1,y1⟩ and end ⟨x2,y2⟩ points may be inset, and the
               // ending line angle may be altered for inset swoopy arrows.
               let x1 = X1[i], y1 = Y1[i], x2 = X2[i], y2 = Y2[i];
-              let lineAngle = Math.atan2(y2 - y1, x2 - x1);
               const lineLength = Math.hypot(x2 - x1, y2 - y1);
+              if (lineLength <= insetStart + insetEnd) return null;
```

Comment:
This change seems unrelated to the fix. Also, I’m not sure we should do it? We don’t do it for rules, for example.

Reply:
It's very ugly when connecting two nodes that are not far enough… Without insets, if lineLength === 0 you get NaNs in the path definition.